cdap-api/src/main/java/co/cask/cdap/api/lineage/field/MergeOperation.java

```diff
@@ -0,0 +1,60 @@
+/*
+ * Copyright © 2018 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package co.cask.cdap.api.lineage.field;
+
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.HashSet;
+import java.util.Set;
+
+/**
+ * Represents merge between multiple origins(operations). Outputs of merge operation
+ * are implicit and is union of outputs of all the origins that are being merged.
```

Comment:
thinking more about this, it seems that a list of origins is not sufficient. Like other operations, each contributor to this merge can have fields from different origins.